hub/core/chunk_engine/generator.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+from typing import Generator, Tuple
+
+from hub.core.chunk_engine.exceptions import ChunkGeneratorError
+
+
+def chunk(
+    content_bytes: bytes,
+    chunk_size: int,
+    last_chunk_num_bytes: int = None,
+) -> Generator[Tuple[bytes, int], None, None]:
+    """
+    Generator function that chunks bytes.
+
+    Chunking is the process of taking the input `content_bytes` & breaking it up into a sequence of smaller bytes called "chunks".
+    The sizes of each chunk are <= `chunk_size`.
+
+    Example 1 (Perfect Fit):
+        content_bytes = b"1094jfnv841q"
+        chunk_size = 4
+        output_chunks = [b"1094", b"jfnv", b"841q"]
```

Comment:
FYI, there is a standard way to represent a _byte_ in this format. It looks like `b"\x00\xFF" where the `\x` escapes the hex code for the byte. You don't have to use it and frankly it's a bit much, but just in case you didn't know.